tasks/sep/base-001-MCP-L1/VALIDATION.md

```diff
@@ -0,0 +1,341 @@
+
+State Changes:
+- **Key:** `0x0000000000000000000000000000000000000000000000000000000000000000` <br/>
+  **Before:** `0x00000000000000000000fd0bf71f60660e2f608ed56e1659c450eb1131200002` <br/>
+  **After:** `0x0000000000000000000000000000000000000000000000000000000000000001` <br/>
+  **Meaning:** The `initialized` boolean is set to `true`. The correctness of
+   this slot is attested to in the Optimism repo at [storageLayout/OptimismMintableERC20Factory.json](https://github.com/ethereum-optimism/optimism/blob/e6ef3a900c42c8722e72c2e2314027f85d12ced5/packages/contracts-bedrock/snapshots/storageLayout/OptimismMintableERC20Factory.json#L2-L22).
+   This state diff will only appear in contracts that were previously not initializable. Other contracts may be reinitialized but it does not show in the state diff because the storage diff is a noop.
+
+- **Key:** `0x0000000000000000000000000000000000000000000000000000000000000001` <br/>
+  **Before:** `0x0000000000000000000000000000000000000000000000000000000000000000` <br/>
+  **After:** `0x000000000000000000000000fd0bf71f60660e2f608ed56e1659c450eb113120` <br/>
+  **Meaning:** Sets `bridge` at slot `0x01` (1). The correctness of
+   this slot is attested to in the Optimism repo at [storageLayout/OptimismMintableERC20Factory.json](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.3.0/packages/contracts-bedrock/snapshots/storageLayout/OptimismMintableERC20Factory.json#L24-L28). The address of the `L1StandardBridge` should be the value set in the slot, left padded with zero bytes to fill the slot. The address of the [L1StandardBridgeProxy](https://github.com/ethereum-optimism/superchain-registry/blob/a62fd38c9239459b27ea50a2df895e89a3305382/superchain/extra/addresses/sepolia/base.json#L5) can be found in the Superchain Registry.
+
+- **Key:** `0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc` <br/>
+  **Before:** `0x0000000000000000000000002f36789426d2e32d33d2fc2f2a7c06e5b1ed17f6` <br/>
+  **After:** `0x000000000000000000000000e01efbeb1089d1d1db9c6c8b135c934c0734c846` <br/>
+  **Meaning:** Implementation address is set to the new `OptimismMintableERC20Factory` implementation. The address can be found in the [Superchain Registry](https://github.com/ethereum-optimism/superchain-registry/blob/5ad42cbb49472a0bf164ade976426f7526ee6dfe/superchain/implementations/networks/sepolia.yaml#L10).
+
+### `0x5D335Aa7d93102110879e3B54985c5F08146091E` (`L1CrossDomainMessengerProxy`)
+
+todo: actual state changes 4 vs 3 in the template
+
+Links:
+- [Etherscan](https://sepolia.etherscan.io/address/0x5D335Aa7d93102110879e3B54985c5F08146091E)
+- [Superchain Registry](https://github.com/ethereum-optimism/superchain-registry/blob/5ad42cbb49472a0bf164ade976426f7526ee6dfe/superchain/extra/addresses/sepolia/metal.json#L3)
+
+State Changes:
+todo: is this one set to true? value doesn't seem to indicate that
```

Comment:
The linked storage layout for slot 0 is
```json
  {
    "bytes": "20",
    "label": "spacer_0_0_20",
    "offset": 0,
    "slot": "0",
    "type": "address"
  },
  {
    "bytes": "1",
    "label": "_initialized",
    "offset": 20,
    "slot": "0",
    "type": "uint8"
  },
  {
    "bytes": "1",
    "label": "_initializing",
    "offset": 21,
    "slot": "0",
    "type": "bool"
  },
  ```
  So the right most 20 bytes are a spacer, and the initialized byte comes after. The old version of this contract probably had `reinitializer(2)`, like the prior ones, which would explain the value there before. But let's confirm the prior value

Reply:
[Comparing](https://github.com/ethereum-optimism/optimism/compare/a7ff5a811612fa338d0a6d6dd72dc2ec9badef6d..e6ef3a900c42c8722e72c2e2314027f85d12ced5) the Base Sepolia deployment with the MCP contracts, it does show the `reinitializer(2)` is present in the initial deployment.